File: tests/zig/fib_recursive.zig
const std = @import("std");
const profile = @import("profile.zig");

var result: i32 = 0;
var depth: i32 = 31;

noinline fn fibR(n: i32) i32 {
    if (n < 2) return n;
    return fibR(n - 1) + fibR(n - 2);
}

fn run(_: void) void {
    result = fibR(@as(*volatile i32, &depth).*);
    std.mem.doNotOptimizeAway(result);
}

pub fn main(init: std.process.Init) !void {
    try profile.profile(init.io, "fibonacci recursive", run, {});
    if (result != 1346269) profile.fail("fibonacci recursive: wrong result");
}

File: tests/cpp/fib_recursive.cpp
#include "profile.h"

static volatile int depth = 31;

PROFILE_NOINLINE int fibR(int n) {
    if (n < 2) return n;
    return fibR(n - 1) + fibR(n - 2);
}

int main() {
    int result = 0;
    profile::run("fibonacci recursive", [&] { result = fibR(depth); profile::keep(result); });
    if (result != 1346269) profile::fail("fibonacci recursive: wrong result");
    return 0;
}
